modules/nf-core/custom/matrixfilter/templates/matrixfilter.R

```diff
@@ -152,11 +154,29 @@ if ((opt\$sample_file != '') && ( ! is.null(opt\$grouping_variable))){
     opt\$minimum_samples <- ncol(abundance_matrix) * opt\$minimum_proportion
 }
 
-# Generate a boolean vector specifying the features to retain
+# Also set up filtering for NAs; use by default minimum_proportion_not_na; only
+# use minimum_samples_not_na if it is provided (default NULL)
 
-keep <- apply(abundance_matrix, 1, function(x){
-    sum(x > opt\$minimum_abundance) >= opt\$minimum_samples
-})
+if (is.null(opt\$minimum_samples_not_na)) {
+    opt\$minimum_samples_not_na <- ncol(abundance_matrix) * opt\$minimum_proportion_not_na
+}
+
+# Define the tests
+
+tests <- list(
+    'abundance' = function(x) sum(x > opt\$minimum_abundance, na.rm = T) >= opt\$minimum_samples,
+    'na' = function(x) !any(is.na(x)) || sum(!is.na(x))/length(x) >= opt\$minimum_samples_not_n
```

Comment:
shouldn't this be 
`sum(!is.na(x))/length(x) >= opt\$minimum_proportion_not_na`
or
`sum(!is.na(x)) >= opt\$minimum_samples_not_na`

Reply:
Aaaaah you are right! What a silly oversight -.- Thanks for pointing that out, I fixed that here: https://github.com/nf-core/modules/pull/3912